components/net/http_loader.rs

```diff
@@ -80,6 +95,7 @@ impl HttpState {
             auth_cache: RwLock::new(AuthCache::new()),
             history_states: RwLock::new(HashMap::new()),
             http_cache: RwLock::new(HttpCache::new()),
+            http_cache_state: (Mutex::new(HashMap::new()), Condvar::new()),
```

Comment:
One thing to note here is that since we use a single condvar for all values in the map, it is possible for this to result in spurious wake-ups of threads that are waiting for a different key to switch state. The behavior will remain correct, and it's rather just a pity to be waking up threads spuriously. 

One benefit of the alternative approach at https://github.com/servo/servo/pull/24203 using cache entries, is that since the locking is internal to the entry, and fetches don't operate on the same entry unless they're for the same url, this problem is avoided naturally. 

With the current design, with the locking being external, it's not clear to me how we can avoid spurious wake-ups.